lib/services/api/api_service.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:survey/gen/configs.gen.dart';
+import 'package:survey/services/http/http_service.dart';
+import 'package:object_mapper/object_mapper.dart';
+import 'package:survey/services/locator/locator_service.dart';
+
+part 'api_exception.dart';
+
+part 'api_params.dart';
+part 'api_service_register.dart';
+
+abstract class ApiService {
+  Future<T> call<T extends Mappable>({
+    required HttpMethod method,
+    String? baseUrl,
+    required String endPoint,
+    ApiParams? params,
+    String? token,
+  });
+
+  void configureGlobalBaseUrl(String? baseUrl);
+
+  void configureGlobalToken(String? token, String? tokenType);
+}
+
+class ApiServiceImpl implements ApiService {
+  static String? _fallbackBaseUrl;
+  static String? _fallbackToken;
+  static String _fallbackTokenType = "Bearer";
+  final HttpService _httpService = locator.get();
+
+  @override
+  Future<T> call<T extends Mappable>({
+    required HttpMethod method,
+    String? baseUrl,
+    required String endPoint,
+    ApiParams? params,
+    String? token,
+    String? tokenType,
+  }) async {
+    final String? finalBaseUrl = baseUrl ?? _fallbackBaseUrl;
+    assert(finalBaseUrl != null);
+
+    final Map<String, dynamic> headers = {};
+
+    // Append bearer token
+    final String? finalToken = token ?? _fallbackToken;
+    if (finalToken != null) {
+      final finalTokenType = tokenType ?? _fallbackTokenType;
+      headers["authorization"] = "$finalTokenType $finalToken";
+    }
+
+    final url = finalBaseUrl! + endPoint;
+    try {
+      final response = await _httpService.request(
+        method: method,
+        data: params?.toJson(),
+        url: url,
+        headers: headers,
+      ) as Map<String, dynamic>;
+      return Mapper.fromJson(
+              response["data"]["attributes"] as Map<String, dynamic>)
+          .toObject<T>();
+    } on HttpException catch (e) {
+      throw ApiException.fromHttpException(e) ?? e;
+    }
+  }
+
+  @override
+  void configureGlobalBaseUrl(String? baseUrl) {
+    _fallbackBaseUrl = baseUrl;
+  }
```

Comment:
the method names here is a bit confusing as you're calling `configure Global var` but inside, you're setting the `fallback var` 🤔  should we name it something more generic or would you consider changing the var names instead? something like `_baseUrl` could be sufficient imo instead of `_fallback....`? Same with the below `token`.
Or maybe you want to call the methods name here as just `configureBaseUrl`? 🤔